### PR TITLE
Fd 1205 thread registry

### PR DESCRIPTION
**Pull request by stackdump**
POC for thread registry

I'm going to try to install it to current codebase - but this is what it looks like sofar

### CODE REVIEW

File: registry/registry.go [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+			Logger: log.ThreadLogger,
+		}
+
+		func() { // add locals to registry
+			StoreManager.Mutex.Lock()
+			defer StoreManager.Mutex.Unlock()
+			StoreManager.Index[gid] = l
+		}()
+
+		<- start // synchronize thread starts
```
Comment (factom-clay) [inline registry/registry.go]: Who sends the start signals? Are you faking a wait group? would a  wait group be better? Do you have test code for this ?

Comment (factom-clay) [reply registry/registry.go]: I see, you close the start channel and let the nil read start everyone.

File: log/logger_test.go [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package log_test
+
+import (
+	"github.com/FactomProject/factomd/log"
+	"testing"
+)
+
+func TestLogPrintf(t *testing.T) {
+	log.LogPrintf("testing", "unittest %v", "FOO")
```
[the remark this answers is not in the document]
Comment (stackdump) [reply log/logger_test.go]: I guess I'll assert the file is created

File: registry/registry.go (hunk 1 above)
Comment (stackdump) [reply registry/registry.go]: maybe a wait group would be appropriate
is there a specific scenario you have in mind?

Comment (stackdump) [reply registry/registry.go]: I guess I can use one to add a - 'after all threads exit' behavior

**Comment by stackdump** [comment]
rebuilding to use wait groups + will integrate w/ existing codebase

**Pull request closed without merging**